tests/Sniffs/Complexity/CognitiveSniffTest.php

```diff
@@ -0,0 +1,74 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Sniffs\Complexity;
+
+use Generator;
+use SlevomatCodingStandard\Sniffs\TestCase;
+use function file_get_contents;
+use function glob;
+use function key;
+use function preg_match;
+use function sprintf;
+
+class CognitiveSniffTest extends TestCase
+{
+
+	public function dataProviderFiles(): Generator
+	{
+		$files = glob(__DIR__ . '/data/cognitive/*.php');
+		if ($files === false) {
+			$files = [];
+		}
+		foreach ($files as $filepath) {
+			$contents = (string) file_get_contents($filepath);
+			preg_match('/^#\[Complexity\((\d+)\)\]$/m', $contents, $matches);
+			$complexity = $matches[1];
+			preg_match('/function (\w+)/', $contents, $matches);
+			$functionName = $matches[1];
```

Comment:
And create the list of function names, complexity and expected line error (see below).